### PR TITLE
Split fixtures

### PR DESCRIPTION
**Pull request by abdounikarim**
Hello,

I just split fixtures into multiples files. 😉

### CODE REVIEW

File: src/DataFixtures/AppFixtures.php [hunk 1]
```diff
@@ -11,92 +11,22 @@
...
 use Symfony\Component\String\Slugger\SluggerInterface;
-use function Symfony\Component\String\u;
 
-class AppFixtures extends Fixture
+abstract class AppFixtures extends Fixture
 {
-    private $passwordEncoder;
-    private $slugger;
+    protected $passwordEncoder;
+    protected $slugger;
```
Comment (bocharsky-bw) [inline src/DataFixtures/AppFixtures.php]: We only need this in `PostFixtures`

File: src/DataFixtures/AppFixtures.php [hunk 2]
```diff
@@ -11,92 +11,22 @@
...
 use Symfony\Component\Security\Core\Encoder\UserPasswordEncoderInterface;
 use Symfony\Component\String\Slugger\SluggerInterface;
-use function Symfony\Component\String\u;
 
-class AppFixtures extends Fixture
+abstract class AppFixtures extends Fixture
 {
-    private $passwordEncoder;
-    private $slugger;
+    protected $passwordEncoder;
```
Comment (bocharsky-bw) [inline src/DataFixtures/AppFixtures.php]: We only need this in `UserFixtures`

File: src/DataFixtures/AppFixtures.php [hunk 3]
```diff
@@ -11,92 +11,22 @@
...
-            }
-
-            $manager->persist($post);
-        }
-
-        $manager->flush();
-    }
-
-    private function getUserData(): array
+    protected function getUserData(): array
```
Comment (bocharsky-bw) [inline src/DataFixtures/AppFixtures.php]: We only need this in `UserFixtures`

File: src/DataFixtures/AppFixtures.php [hunk 4]
```diff
@@ -121,27 +51,7 @@ private function getTagData(): array
...
-                $this->getReference(['jane_admin', 'tom_admin'][0 === $i ? 0 : random_int(0, 1)]),
-                $this->getRandomTags(),
-            ];
-        }
-
-        return $posts;
-    }
-
-    private function getPhrases(): array
+    protected function getPhrases(): array
```
Comment (bocharsky-bw) [inline src/DataFixtures/AppFixtures.php]: We only need this in `PostFixtures`

File: src/DataFixtures/AppFixtures.php [hunk 5]
```diff
@@ -177,20 +87,7 @@ private function getPhrases(): array
...
-        do {
-            $text = u('. ')->join($phrases)->append('.');
-            array_pop($phrases);
-        } while ($text->length() > $maxLength);
-
-        return $text;
-    }
-
-    private function getPostContent(): string
+    protected function getPostContent(): string
```
Comment (bocharsky-bw) [inline src/DataFixtures/AppFixtures.php]: We only need this in `PostFixtures`

**Comment by javiereguiluz** [comment]
@abdounikarim we thank you this contribution ... but sadly we must close it without merging. The reason is that this Demo app follows the Symfony best practices strictly. Regarding fixtures, this is [what we recommend](https://symfony.com/doc/current/bundles/DoctrineFixturesBundle/index.html#splitting-fixtures-into-separate-files):

fixtures-multiple-files

So, we must keep the fixtures in a single file ... even if it's a slightly long class. Thanks for understanding 🙏

**Pull request closed without merging**